packages/language-server/src/plugins/typescript/features/SemanticTokenProvider.ts

```diff
@@ -28,7 +28,10 @@ export class SemanticTokensProviderImpl implements SemanticTokensProvider {
 			filePath,
 			{
 				start,
-				length: range ? fragment.offsetAt(fragment.getGeneratedPosition(range.end)) - start : fragment.text.length,
+				length: range
+					? fragment.offsetAt(fragment.getGeneratedPosition(range.end)) - start
+					: // We don't want tokens for things added by astro2tsx
+					  fragment.text.lastIndexOf('export default function (_props:') || fragment.text.length,
```

Comment:
Previously we'd generate tokens for code generated by `astro2tsx`, which I don't think would create issues? But is still technically improper behaviour